Source/Core/Resource.js

```diff
@@ -1755,26 +1825,50 @@ define([
      */
     Resource._Implementations = {};
 
-    Resource._Implementations.createImage = function(url, crossOrigin, deferred) {
-        var image = new Image();
+    Resource._Implementations.createImage = function(url, crossOrigin, deferred, flipY) {
+        if (!FeatureDetection.supportsCreateImageBitmap()) {
+            var image = new Image();
 
-        image.onload = function() {
-            deferred.resolve(image);
-        };
+            image.onload = function() {
+                deferred.resolve(image);
+            };
 
-        image.onerror = function(e) {
-            deferred.reject(e);
-        };
+            image.onerror = function(e) {
+                deferred.reject(e);
+            };
 
-        if (crossOrigin) {
-            if (TrustedServers.contains(url)) {
-                image.crossOrigin = 'use-credentials';
-            } else {
-                image.crossOrigin = '';
+            if (crossOrigin) {
+                if (TrustedServers.contains(url)) {
+                    image.crossOrigin = 'use-credentials';
+                } else {
+                    image.crossOrigin = '';
+                }
             }
+
+            image.src = url;
+        } else {
+            // createImageBitmap will block the main thread during decode if a blob is not supplied.
+            Resource.fetchBlob({
+                url: url
+            }).then(function(blob) {
+                Resource._Implementations.createImageBitmapFromBlob(blob, deferred, flipY);
+            }).otherwise(deferred.reject);
         }
+    };
 
-        image.src = url;
+    Resource._Implementations.createImageBitmapFromBlob = function(blob, deferred, flipY) {
```

Comment:
Already mentioned this above, but there's no reason to take a deferred here, just return the promise you create below.  In general, avoid deferred.